dex/farm/src/lib.rs

```diff
@@ -493,65 +492,67 @@ pub trait Farm:
         farming_amount: &BigUint,
         destination: &ManagedAddress,
         opt_accept_funds_func: &OptionalArg<ManagedBuffer>,
-    ) -> SCResult<()> {
+    ) {
         self.transfer_execute_custom(
             destination,
             farming_token_id,
             0,
             farming_amount,
             opt_accept_funds_func,
-        )?;
-        Ok(())
+        )
+        .unwrap_or_signal_error(self.type_manager());
     }
 
-    fn send_rewards(
-        &self,
-        reward_token_id: &TokenIdentifier,
-        reward_amount: &BigUint,
-        destination: &ManagedAddress,
-        opt_accept_funds_func: &OptionalArg<ManagedBuffer>,
-    ) -> SCResult<()> {
-        if reward_amount > &mut 0 {
+    fn send_rewards(&self, context: &mut GenericContext<Self::Api>) {
+        if context.get_position_reward().unwrap() > &0u64 {
             self.transfer_execute_custom(
-                destination,
-                reward_token_id,
+                context.get_caller(),
+                context.get_reward_token_id(),
                 0,
-                reward_amount,
-                opt_accept_funds_func,
-            )?;
+                context.get_position_reward().unwrap(),
+                context.get_opt_accept_funds_func(),
+            )
+            .unwrap_or_signal_error(self.type_manager());
         }
-        Ok(())
+
+        context.set_final_reward(self.create_payment(
+            context.get_reward_token_id(),
+            0,
+            context.get_position_reward().unwrap(),
+        ));
     }
 
     #[view(calculateRewardsForGivenPosition)]
     fn calculate_rewards_for_given_position(
         &self,
         amount: BigUint,
         attributes: FarmTokenAttributes<Self::Api>,
```

Comment:
add a TODO here to refactor this in order to send the farm token attributes as exact values.

Reply:
when changing the farm token attributes, this function signature will change also. Also this function would receive a huge load of parameters, so i'd rather keep it contained by using the struct